libraries/RW/GCP/OpsSuite.py

```diff
@@ -49,6 +50,30 @@ def get_credentials(self) -> object:
         """
         return self._credentials
 
+    def get_token(self, gcp_credentials : platform.Secret=None) -> object:
+        """
+        Retrieve short lived bearer token from service account authentication.
+
+        If used directly in a codebundle, should use `Set Log Level None` to keep sensitive tokens out of logs. 
+
+        Examples:
+        | RW.GCP.OpsSuite.Get Token  | gcp_credentials=${ops-suite-sa}
+        Return Value:
+        | The access token, good for 3600s.   |
+        """
+        if not gcp_credentials:
+            raise ValueError(f"service_account is empty")
+        sa = json.loads(gcp_credentials.value, strict=False)
+        creds = service_account.Credentials.from_service_account_info(sa, scopes=['https://www.googleapis.com/auth/cloud-platform'])
+        
+        # See https://cloud.google.com/docs/authentication/token-types#access-contents
+        # Access tokens are by default good for 1 hour / 3600 seconds 
+        # https://github.com/googleapis/google-auth-library-python/blob/main/google/oauth2/service_account.py
+
+        request = google.auth.transport.requests.Request()
+        creds.refresh(request)
+        return creds.token
```

Comment:
From my prior comment, it would be:
1: return platform.Secret(key=token, val=creds.token)
2: a new func which takes the token secret and produces the headers like you're doing in robot, and it itself also returns a platform.Secret

Reply:
Further clarification: you'd use the new function in place of RW.Utils.Create Secret on line 19, so the token value never needs to be accessed in the robot layer, the python layer would access it, create a new header secret and return that.

Reply:
Should be resolved by https://github.com/runwhen-contrib/rw-public-codecollection/pull/32/commits/4313fb30c818f85e637e23618602864281a5233b